### PR TITLE
Minor cleanup (doc comments etc.)

### PR DESCRIPTION
**Pull request by forty8bits**
Nothing major here, just a cleanup of the doc comments as well as some minor additions to the code (no functionality has been changed). No longer using unnecessary async $fh.data() wrapper around localStorage.

### CODE REVIEW

File: fhactproxy.js [hunk 1]
```diff
@@ -1,61 +1,76 @@
...
-     * Name of the serverside function to call
-     */
-    act: undefined,
+        // The name of the act method to call.
+        act: null,
+
+        // The requests body which will be sent to the act call (must be an object).
+        req: null,
+
+        // We don't want to include some of the default parameters along with our request.
```
Comment (forty8bits) [inline fhactproxy.js]: Honestly, I'm not sure if including these is absolutely necessary, but from reading the source of the Server proxy it seemed like it would be nice to be explicit here.

File: fhactproxy.js [hunk 2]
```diff
@@ -1,61 +1,76 @@
...
+        startParam: false,
+        limitParam: false,
+        groupParam: false,
+        sortParam: false,
+        filterParam: false,
+        directionParam: false,
+        enablePagingParams: false,
+
+        // The default timeout of 30 seconds is far too long for a good mobile user experience.
+        timeout: 15000
```
Comment (forty8bits) [inline fhactproxy.js]: The default timeout has been halved, but remains overridable by the user... in my opinion 15 seconds may even be too high a default to allow for good user experiences within mobile apps.

File: fhactproxy.js [hunk 3]
```diff
@@ -1,61 +1,76 @@
...
-          res = res.data || res.records;
+        function doAct(actId, req) {
+            $fh.act({
+                    'act': actId,
+                    'req': req
+                },
+                function(res) {
+
+                    // Check for a 'data' or 'records' property if what we got back isn't an array.
+                    if (!Array.isArray(res)){
```
Comment (forty8bits) [inline fhactproxy.js]: Using the Array.isArray() method here just for explicitness, as well as the fact that (for whatever ridiculous reason) there's nothing stopping someone from assigning a 'length' property within a plain object.

File: fhactproxy.js (hunk 1 above)
Comment (cianclarke) [inline fhactproxy.js]: Added commets good, but you've nuked the JSDuck style comment blocks, like these?
@property {Object} act
Rather they were left in

File: fhactproxy.js (hunk 2 above)
Comment (cianclarke) [inline fhactproxy.js]: The default params you've explicitly made false - not sure what they all do, and would rather trust default behaviour for now, unless you've thoroughly tested this in a few scenarios and on a few devices

Comment (cianclarke) [inline fhactproxy.js]: Also, default timeout of 30 seconds is fine, 15 seconds is too short for the legacy backend systems we sometimes have to integrate with. If anything it'd be good to increase this...

File: fhactproxy.js (hunk 3 above)
Comment (cianclarke) [inline fhactproxy.js]: Array.isArray() is ES5 only, not supported in earlier browsers & not suitable for mobile.

**Comment by cianclarke** [comment]
Thanks for the pull request Gareth - looks good, a few small changes before we merge this!

File: fhactproxy.js (hunk 1 above)
Comment (forty8bits) [inline fhactproxy.js]: Yeah I was actually thinking that those property tags were erroneously placed because I tend to use the standard JSDoc syntax (in which property tags are defined within a main class comment)... I'll add them back in. Might also be worth noting that JSDuck has extensive auto-detection of these tags, and according to https://github.com/senchalabs/jsduck/wiki/@property we can cut these comments down to just the description (although I tend to like being explicit anyway). I'll add them back in and see about adding even more docs.

File: fhactproxy.js (hunk 2 above)
Comment (forty8bits) [inline fhactproxy.js]: I haven't taken the time to check them out either; I simply put them in because the source code documentation within the Server proxy class stated that these parameters are sent with the request by default unless set to false explicitly... I'd like to do more testing out of curiosity, but we can surely remove them without any negative effect.

As for the default timeout, point taken about the legacy systems. I'll reset it back to 30, as upsetting as that'll be. God help anybody with an app that takes over a few seconds for any request though! :P

File: fhactproxy.js (hunk 3 above)
Comment (forty8bits) [inline fhactproxy.js]: Yeah I tend to use a lot of ES5 methods as of late, given that most mobile browsers have very good support for most of them (especially the WebKit based ones, which Sencha specifically targets). I know this method is supported on Android 1.5+ (so fine for all) and iOS 4+ (maybe not acceptable?).

In any case, I still think just checking for length isn't good enough, for the reasons above. I say we use underscores method here:

```
_.isArray = nativeIsArray || function(obj) {
    return toString.call(obj) == '[object Array]';
};
```

**Comment by forty8bits** [comment]
No worries, all good comments. I'll get it fixed up and file another pull request. It just occurred to me that the code style I'm using actually doesn't comply to our style guide anyway (4 spaces rather than 2 etc, hadn't set it up in WebStorm on the new Mac yet!) so I'll sort that out while I'm at it.

File: fhactproxy.js [hunk 4]
```diff
@@ -1,61 +1,76 @@
...
+            localItem;
+
+        // By convention if we get a string as the request, it's implied that we should check
+        // localStorage for a matching request object.
+        if (typeof req === 'string') {
+            localItem = localStorage.getItem(req);
+
+            // localStorage will return null if the key doesn't exist, so we must normalise it.
+            if (!localItem) localItem = '';
+            localItem = JSON.parse(localItem);
```
Comment (forty8bits) [inline fhactproxy.js]: This could actually be made more concise:

`localItem = !localItem ? {} : JSON.parse(localItem);`

File: fhactproxy.js [hunk 5]
```diff
@@ -1,61 +1,69 @@
...
+     * @property {string} act
+     * The name of the act method to call.
      */
-    act: undefined,
+    act: null,
 
     /**
-     * @property {Object} req
-     * Request body to be sent with any act call
+     * @property {Object/string} req
```
Comment (forty8bits) [inline fhactproxy.js:20]: Restored the JSDuck docs. I like to stick to a certain style (which is accepted by both Closure Compiler and JSDuck), in which I use lowercase for the JS primitives and capitalise the first letter only for non-primitives. This is the way it tends to be done within Closure Compiler examples, and JSDuck also outlines it as acceptable.

File: fhactproxy.js [hunk 6]
```diff
@@ -1,61 +1,69 @@
...
+
+    function doAct(actId, req) {
+      $fh.act({
+          'act': actId,
+          'req': req
+        },
+        function (res) {
+
+          // Check for a 'data' or 'records' property if what we got back isn't an array.
+          if (toString.call(res) !== '[object Array]') {
```
Comment (forty8bits) [inline fhactproxy.js]: Removed the ES5 specific Array.isArray and replaced it with the method used by Underscore (and afaik most other libraries) to detect Arrays.

File: fhactproxy.js [hunk 7]
```diff
@@ -1,61 +1,69 @@
...
+      req = this.getReq(),
+      localItem;
+
+    // By convention if we get a string as the request, it's implied that we should check
+    // localStorage for a matching request object.
+    if (typeof req === 'string') {
+      localItem = localStorage.getItem(req);
+
+      // localStorage will return null if the key doesn't exist, so we must normalise it.
+      localItem = !localItem ? {} : JSON.parse(localItem);
```
Comment (forty8bits) [inline fhactproxy.js]: It just struck me that the localItem here, if present, is never even used. Needs to be fixed before we merge, stupid mistake on my behalf.

**Comment by camilamacedo86** [comment]
Closing it because it has been open since 19 Sep 2012.
